sklearn/experimental/_column_transformer.py

```diff
@@ -0,0 +1,558 @@
+"""
+The :mod:`sklearn.experimental._column_transformer` module implements utilities
+to work with heterogeneous data and to apply different transformers to
+different columns.
+"""
+# Author: Andreas Mueller
+#         Joris Van den Bossche
+# License: BSD
+
+
+import numpy as np
+from scipy import sparse
+
+from ..base import clone, TransformerMixin
+from ..externals.joblib import Parallel, delayed
+from ..externals import six
+from ..pipeline import (
+    _fit_one_transformer, _fit_transform_one, _transform_one, _name_estimators)
+from ..utils import Bunch
+from ..utils.metaestimators import _BaseComposition
+from ..utils.validation import check_is_fitted
+
+
+__all__ = ['ColumnTransformer', 'make_column_transformer']
+
+
+_ERR_MSG_1DCOLUMN = ("1D data passed to a transformer that expects 2D data. "
+                     "Try to specify the column selection as a list of one "
+                     "item instead of a scalar.")
+
+
+class ColumnTransformer(_BaseComposition, TransformerMixin):
+    """Applies transformers to columns of an array or pandas DataFrame.
+
+    EXPERIMENTAL
+
+    This estimator allows different columns or column subsets of the input
+    to be transformed separately and the results combined into a single
+    feature space.
+    This is useful for heterogeneous or columnar data, to combine several
+    feature extraction mechanisms or transformations into a single transformer.
+
+    Read more in the :ref:`User Guide <column_transformer>`.
+
+    Parameters
+    ----------
+    transformers : list of tuples
+        List of (name, transformer, column) tuples specifying the transformer
+        objects to be applied to subsets of the data.
+
+        name : string
+            Like in FeatureUnion and Pipeline, this allows the transformer and
+            its parameters to be set using ``set_params`` and searched in grid
+            search.
+        transformer : estimator or None
+            If None, nothing is output for this triple. Estimator must support
+            `fit` and `transform`.
+        column : string or int, array-like of string or int, slice or boolean \
+ mask array
+            Indexes the data on its second axis. Integers are interpreted as
+            positional columns, while strings can reference DataFrame columns
+            by name.  A scalar string or int should be used where
+            ``transformer`` expects X to be a 1d array-like (vector),
+            otherwise a 2d array will be passed to the transformer.
+
+    passthrough : string or int, array-like of string or int, slice or \
```

Comment:
I don't think we should support a single string or into here, because it provides no benefit, is treated identically to a list of length 1, and makes for ambiguity with 'remainder'

Reply:
Yes, I actually already raise an explicit error message in this case (see the code in `_validate_passthrough`), but forgot to update the docstring here.